infra/build/functions/main.py

```diff
@@ -0,0 +1,90 @@
+# Copyright 2020 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+################################################################################
+"""Cloud functions for build scheduling"""
+
+import re
+
+from github import Github
+from google.cloud import ndb
+
+VALID_PROJECT_NAME = re.compile(r'^[a-zA-Z0-9_-]+$')
+
+
+# pylint: disable=too-few-public-methods
+class Project(ndb.Model):
+  """Datastore Entity Project"""
+  name = ndb.StringProperty()
+
+
+# pylint: disable=too-few-public-methods
+class GitAuth(ndb.Model):
+  """Datastore Entity GitAuth"""
+  access_token = ndb.StringProperty()
+
+
+def sync_projects(projects):
+  """Sync projects with cloud datastore"""
+  project_query = Project.query()
+  projects_to_remove = [
+      project.key for project in project_query if project.name not in projects
+  ]
+
+  ndb.delete_multi(projects_to_remove)
+
+  existing_projects = {project.name for project in project_query}
+
+  for project in projects:
```

Comment:
nit: do this instead. 

```python
new_projects = [project for project in projects if project not in existing_projects]
ndb.put_multi(new_projects)
```

Reply:
`new_projects = [
      Project(name=project)
      for project in projects
      if project not in existing_projects
  ]
  ndb.put_multi(new_projects)`

Since ndb.put_multi require entities changing it to this

Reply:
Yep, that's the right one :)